src/modules/endpoint/DataContainer.js

```diff
@@ -0,0 +1,250 @@
+// @flow
+
+import {
+  CategoriesMapModel,
+  CityModel, EventModel,
+  LanguageModel
+} from '@integreat-app/integreat-api-client'
+import DatabaseContext from './DatabaseContext'
+import type { ResourceCacheStateType } from '../app/StateType'
+import DatabaseConnector from './DatabaseConnector'
+
+interface DataContainerInterface {
+  /**
+   * Changes the context to the supplied city-language combination and loads all corresponding persisted data if
+   * existent. Initializes non persisted fields with null.
+   * @param cityCode
+   * @param languageCode
+   */
+  setContext: (cityCode: string, languageCode: string) => Promise<void>,
+
+  /**
+   * Returns whether the current context matches the given city-language combination or not.
+   * @param cityCode
+   * @param languageCode
+   */
+
+  hasContext: (cityCode: string, languageCode: string) => boolean,
```

Comment:
@Schedulaar I added this again for performance reasons. The way we did it without it took about 1 second each time the page was changed.

Reply:
then there's something wrong with the rest of the code :D

Reply:
The thing is that all the other sagas are executed to the point where the code checks whether the corresponding data has already been loaded or not. I think all the generator-saga-thing just takes a bit.

Reply:
Nvm, the wrong `xyzLoaded()` functions caused the delay as discussed.